crates/services/consensus_module/poa/src/service.rs

```diff
@@ -137,40 +142,69 @@ pub struct Task<T, B, I> {
     /// Deadline clock, used by the triggers
     timer: DeadlineClock,
     consensus_params: ConsensusParameters,
+    sync_state: tokio::sync::watch::Receiver<SyncState>,
+    sync_task_handle: ServiceRunner<SyncTask>,
 }
 
-impl<T, B, I> Task<T, B, I>
+impl<T, B, I> MainTask<T, B, I>
 where
     T: TransactionPool,
+    I: BlockImporter,
 {
-    pub fn new(
+    pub fn new<P: P2pPort>(
         last_block: &BlockHeader,
         config: Config,
         txpool: T,
         block_producer: B,
         block_importer: I,
+        p2p_port: P,
     ) -> Self {
         let tx_status_update_stream = txpool.transaction_status_events();
         let (request_sender, request_receiver) = mpsc::channel(100);
         let last_timestamp = last_block.time();
         let duration =
             Duration::from_secs(Tai64::now().0.saturating_sub(last_timestamp.0));
         let last_block_created = Instant::now() - duration;
+
+        let min_connected_reserved_peers = config.min_connected_reserved_peers;
+        let time_until_synced = config.time_until_synced;
+
+        let initial_sync_state =
```

Comment:
It should be the internal logic of the `SyncTask`. The `sync_state` can be part of the `SyncTask::SharedState` and you can subscribe after creation fo the `SyncTask `=)